src/domain/position/binProcessor.js

```
@@ -0,0 +1,242 @@
+
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ * INFITX
+ - Vijay Kumar Guthi <vijaya.guthi@infitx.com>
+
+ --------------
+ ******/
+'use strict'
+
+const Logger = require('@mojaloop/central-services-logger')
+const BatchPositionModel = require('../../models/position/batch')
+const BatchPositionModelCached = require('../../models/position/batchCached')
+const PositionPrepareDomain = require('./prepare')
+const SettlementModelCached = require('../../models/settlement/settlementModelCached')
+const Enum = require('@mojaloop/central-services-shared').Enum
+const ErrorHandler = require('@mojaloop/central-services-error-handling')
+// TODO: We may not need this if we optimize the participantLimit query
+const participantFacade = require('../../models/participant/facade')
+
+/**
+ * @function processBins
+ *
+ * @async
+ * @description This is the domain function to process a list bins containing position messages grouped by participant account.
+ *
+ * @param {array} bins - a list of account-bins to process
+ * @param {object} trx - Database transaction object
+ *
+ * @returns {results} - Returns a list of bins with results or throws an error if failed
+ */
+const processBins = async (bins, trx) => {
+  const transferIdList = []
+  await iterateThroughBins(bins, async (_accountID, _action, item) => {
```

why no errorCb?

---

The callback routine in this case is not expected to cause any exception. `iterateThroughBins` has been updated to throw unexpected errors rather than silence them when `errorCb` is undefined.

---

The `cb` routine is not expected to cause any error that requires handling. `iterateThroughBins` have been updated to log unexpected/unhandled errors rather than swallow them.